src/challenge/Challenge.tsx

```diff
@@ -320,40 +340,69 @@ class Challenge extends React.Component<ChallengeProps, ChallengeState> {
     );
   };
 
+  renderFixedInput = () => {
+    return (
+      <ThemeProvider
+      theme={this.state.theme === "vs-dark" ? darkTheme : pageTheme}
+      >
+        <Box
+          sx={{
+            width: "100%",
+            height: "100%",
+            bgcolor: "background.default",
+          }}
+        >
+
+        <InputBox ref={this.inputBoxRef} />
+
+        </Box>
+      </ThemeProvider>
+    );
+  };  
+
   renderOutput = () => {
-    if (
-      this.props.typ === "canvas" ||
-      this.state.typInferred === ChallengeTypes.TYP_CANVAS
-    ) {
-      return (
-        <ThemeProvider
-          theme={this.state.theme === "vs-dark" ? darkTheme : pageTheme}
+
+    let panes = [
+      {
+        label: "Console",
+        content: this.renderConsole(),
+        show: true
+      },      
+      {
+        label: "Fixed input",
+        content: this.renderFixedInput(),
+        show: this.state.isFixedInput
+      }
+    ];
+
+    if (this.props.typ === "canvas" || this.state.typInferred === ChallengeTypes.TYP_CANVAS) {
+      panes.push({
+        label: "Canvas",
+        content: <CanvasDisplay ref={this.canvasDisplayRef} />,
+        show: true
+      });
+    } 
+
+    return (
+      <ThemeProvider
+        theme={this.state.theme === "vs-dark" ? darkTheme : pageTheme}
+      >
+        <Box
+          sx={{
+            width: "100%",
+            height: "100%",
+            bgcolor: "background.default",
+          }}
         >
-          <Box
-            sx={{
-              width: "100%",
-              height: "100%",
-              bgcolor: "background.default",
-            }}
-          >
-            <TabbedView
-              ref={this.tabbedViewRef}
-              panes={[
-                {
-                  label: "Canvas",
-                  content: <CanvasDisplay ref={this.canvasDisplayRef} />,
-                },
-                {
-                  label: "Console",
-                  content: this.renderConsole(),
-                },
-              ]}
-            />
-          </Box>
-        </ThemeProvider>
-      );
-    }
-    return this.renderConsole();
+                
+        <TabbedView
```

Comment:
can we hide the tab ribbon when there is only one tab (i.e. console?)

Reply:
done